projects/packages/connection/src/rest-endpoints/class-token-endpoints.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Jetpack REST API token endpoints.
+ *
+ * @package automattic/jetpack-connection
+ */
+
+namespace Automattic\Jetpack\Connection\REST_Endpoints;
+
+use Automattic\Jetpack\Connection\Manager;
+use Automattic\Jetpack\Connection\Rest_Authentication;
+use Automattic\Jetpack\Connection\REST_Connector;
+use Automattic\Jetpack\Connection\Tokens;
+use WP_Error;
+use WP_REST_Server;
+
+/**
+ * Jetpack REST API token endpoints.
+ *
+ * @package Automattic\Jetpack\Core_API
+ */
+class Token_Endpoints {
+
+	/**
+	 * Registering endpoints.
+	 */
+	public function register_endpoints() {
+		register_rest_route(
+			'jetpack/v4',
+			'/user-token',
+			array(
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( static::class, 'update_user_token' ),
+					'permission_callback' => array( static::class, 'update_user_token_permission_check' ),
+					'args'                => array(
+						'user_token'          => array(
+							'description' => __( 'New user token', 'jetpack' ),
+							'type'        => 'string',
+							'required'    => true,
+						),
+						'is_connection_owner' => array(
+							'description' => __( 'Is connection owner', 'jetpack' ),
+							'type'        => 'boolean',
+						),
```

Comment:
Do we need mark those arguments as `'required' => true,`?

Reply:
Hi @jeherve.
`user_token` is marked so, and the other one (`is_connection_owner`) is optional.